publication-event-handlers/src/main/java/no/unit/nva/publication/events/handlers/tickets/AcceptedPublishingRequestEventHandler.java

```diff
@@ -178,14 +187,16 @@ private Publication fetchPublication(SortableIdentifier resourceIdentifier) {
      * Creating DoiRequest for a publication necessarily owned by publication owner institution and not the institution
      * that requests the doi.
      *
-     * @param publication to create a DoiRequest for
+     * @param publication  to create a DoiRequest for
+     * @param userInstance
      */
-    private void createDoiRequestIfNeeded(Publication publication) {
+    private void createDoiRequestIfNeeded(Publication publication, UserInstance userInstance) {
         if (hasDoi(publication) && !doiRequestExists(publication)) {
-            attempt(() -> DoiRequest.fromPublication(publication)
-                              .withOwner(publication.getResourceOwner().getOwner().getValue())
-                              .withOwnerAffiliation(publication.getResourceOwner().getOwnerAffiliation())
-                              .persistNewTicket(ticketService)).orElseThrow();
+            var doiRequest = (DoiRequest) DoiRequest.fromPublication(publication)
+                                 .withOwner(publication.getResourceOwner().getOwner().getValue())
+                                 .withOwnerAffiliation(publication.getResourceOwner().getOwnerAffiliation());
+            doiRequest.setTicketEvent(DoiRequestedEvent.create(userInstance, Instant.now()));
+            attempt(() -> doiRequest.persistNewTicket(ticketService)).orElseThrow();
             logger.info(DOI_REQUEST_CREATION_MESSAGE, publication.getIdentifier());
```

Comment:
I wish we had method to create ticket from publication and user, as we do with publication, so there is no need to set things in different places. But it is a larger refactoring :)

Reply:
Create the task, I guess